src/org/parosproxy/paros/core/proxy/ProxyThread.java

```diff
@@ -451,13 +452,19 @@ protected void processHttp(HttpRequestHeader requestHeader, boolean isSecure) th
 		        	return;
 			    }
 			}
-
-			if (parentServer.isEnableApi() &&
-					API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader))) {
-				// It was an API request
-				return;
+			if (parentServer.isEnableApi()) {
+				msg = API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader));
+				if (msg != null) {
+					ZapGetMethod method = (ZapGetMethod) msg.getUserObject();

```

Comment:
Does it worth get/check the method? (It should never be present, the API does not set it?)